Domain/Services/FlightService.cs

```diff
@@ -0,0 +1,70 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+using AirportTicketBookingSystem.Domain.Models;
+using AirportTicketBookingSystem.Domain.Repositories;
+
+namespace AirportTicketBookingSystem.Domain.Services
+{
+    public class FlightService
+    {
+        private readonly FlightRepository _flightRepository;
+
+        public FlightService()
+        {
+            _flightRepository = new FlightRepository();
+        }
+
+        public void AddFlight(Flight flight)
+        {
+            _flightRepository.AddFlight(flight);
+        }
+
+        public List<Flight> SearchFlights(
+            string departureCountry = "",
+            string destinationCountry = "",
+            string departureAirport = "",
+            string arrivalAirport = "",
+            DateTime? departureDate = null,
+            SeatClass? seatClass = null,
+            double? maxPrice = null)
+        {
+            var flights = _flightRepository.GetAllFlights();
+            return flights.Where(f =>
+                (string.IsNullOrEmpty(departureCountry) || f.DepartureCountry.Equals(departureCountry, StringComparison.OrdinalIgnoreCase)) &&
+                (string.IsNullOrEmpty(destinationCountry) || f.DestinationCountry.Equals(destinationCountry, StringComparison.OrdinalIgnoreCase)) &&
+                (string.IsNullOrEmpty(departureAirport) || f.DepartureAirport.Equals(departureAirport, StringComparison.OrdinalIgnoreCase)) &&
+                (string.IsNullOrEmpty(arrivalAirport) || f.ArrivalAirport.Equals(arrivalAirport, StringComparison.OrdinalIgnoreCase)) &&
+                (!departureDate.HasValue || f.DepartureDate.Date == departureDate.Value.Date) &&
+                (maxPrice == null || (seatClass != null && Enum.TryParse(seatClass.ToString(), out SeatClass seatClassEnum) ?
```

Comment:
I see this logic for enum `Enum.TryParse...` used in multiple places, you can make an extension method that would take a string and return it's enum value if applicable, and another method to check if a enum value is valid and defined in the given enum type (use generics).
Using such methods promotes reusability and readability.